ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -106,34 +111,39 @@ public SafeFuture<BlobSidecarsAndValidationResult> getAvailabilityCheckResult()
   @Override
   public BlobSidecarsAndValidationResult validateImmediately(final List<BlobSidecar> blobSidecars) {
 
-    final List<KZGCommitment> kzgCommitments = kzgCommitmentsSupplier.get();
+    final List<KZGCommitment> kzgCommitmentsFromBlock = kzgCommitmentsFromBlockSupplier.get();
 
     if (!blobSidecars.isEmpty()) {
-      return validateBatch(blobSidecars, kzgCommitments);
+      final BlobSidecarsAndValidationResult blobSidecarsAndValidationResult =
+          validateBatch(blobSidecars);
+      performCompletenessValidation(blobSidecarsAndValidationResult);
```

Comment:
You ignore the result of completeness check, there should be probably return on this line

Reply:
`performCompletenessValidation` is designed to throw only

Reply:
It has the line `return BlobSidecarsAndValidationResult.NOT_REQUIRED;`
Unless it's not `void` return I assume we care on return

Reply:
you're right.